packages/edit-site/src/components/style-book/index.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * External dependencies
+ */
+import { colord } from 'colord';
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { Button, TabPanel } from '@wordpress/components';
+import { arrowLeft } from '@wordpress/icons';
+import { __ } from '@wordpress/i18n';
+import {
+	getCategories,
+	getBlockTypes,
+	getBlockFromExample,
+} from '@wordpress/blocks';
+import { BlockPreview } from '@wordpress/block-editor';
+import { Fragment } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { useStyle } from '../global-styles';
+
+export default function StyleBook() {
+	const [ backgroundColor ] = useStyle( 'color.background' );
+	return (
+		<div className="edit-site-style-book">
+			<Button
+				className="edit-site-style-book__back-button"
+				icon={ arrowLeft }
+				onClick={ () => {} }
+			>
+				{ __( 'Back' ) }
+			</Button>
+			<TabPanel
+				className="edit-site-style-book__tab-panel"
+				// TODO: hide categories that have no examples
+				tabs={ getCategories().map( ( category ) => ( {
+					name: category.slug,
+					title: category.title,
+					icon: category.icon,
+				} ) ) }
+			>
+				{ ( tab ) => (
+					<div
+						className={ classnames(
+							'edit-site-style-book__canvas',
+							{
+								'has-dark-background':
+									colord( backgroundColor ).isDark(),
+							}
+						) }
+						style={ {
+							background: backgroundColor,
+						} }
+					>
+						<div className="edit-site-style-book__blocks">
+							{ getBlockTypes()
```

Comment:
Nice. I wonder if there's a way to memoize this based on the activeCategory (`tab.name`) I tried to memoize the `.edit-site-style-book__blocks` element, but it's a bit tough as it's re-rendered inside the tab regardless (and makes it worse 🤣 )

Reply:
Yeah could do. It might make switching to a tab _the second time_ a little bit faster. I'll have a play with performance after we land on a final design.

Reply:
I suspect most of the sluggishness comes from having so many frames and resize listeners 😀 I don't think there's a lot we can do at the React component level to fix that. I did notice that when I add `transition: background .3s` to the canvas that things _feel_ less sluggish, especially in Chrome. It think it's good enough for a v1.

Reply:
Looks good, and agree it's a good candidate for v1. ❤️ 

Should we place it behind an experimental flag?

I wonder if we can get around the iframe issue by rendering the block preview using the [useBlockPreview hook](https://github.com/WordPress/gutenberg/blob/abc8d7e850e0e6162725fe22f1b1d033f3d22ee4/packages/block-editor/src/components/block-preview/index.js#L82-L81) 

I tried it out, basically replicating [this unit test](https://github.com/WordPress/gutenberg/blob/d50e613d110ea7ccd5de3f647c8dcf3b72bdda74/packages/block-editor/src/components/block-preview/test/index.js#L84) and 🤷 

https://user-images.githubusercontent.com/6458278/203697141-16b31f9d-875f-44c7-80f6-4557bdf67b2d.mp4

No iframes at least, but not sure about implications re: styles 😄

Reply:
The iframes are what the styles are loaded into, hence why they appear unstyled for you there. Potentially we could make all of the previews go into a single iframe which should be less intensive on the browser. Can do that as a follow-up if necessary.

Reply:
> Potentially we could make all of the previews go into a single iframe which should be less intensive on the browser

Ah okay, thanks for the explainer. Makes sense.